subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -161,4 +170,34 @@ public boolean isSatisfiedBy(Task element) {
 
         // TODO - check should depend on test
     }
+
+    private void configureTestedComponent(final Project project) {
+        project.getPlugins().withType(SwiftExecutablePlugin.class, new Action<SwiftExecutablePlugin>() {
+            @Override
+            public void execute(SwiftExecutablePlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+
+        project.getPlugins().withType(SwiftLibraryPlugin.class, new Action<SwiftLibraryPlugin>() {
+            @Override
+            public void execute(SwiftLibraryPlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+    }
+
+    private void configureTestedSwiftComponent(Project project) {
+        TaskContainer tasks = project.getTasks();
+        DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
+
+        SwiftCompile compile = tasks.withType(SwiftCompile.class).getByName("compileSwift");
+
+        SwiftCompile compileTest = tasks.withType(SwiftCompile.class).getByName("compileTestSwift");
+        compileTest.dependsOn(compile);
+        compileTest.includes(buildDirectory.dir("main/objs"));
+
+        AbstractLinkTask linkTest = tasks.withType(AbstractLinkTask.class).getByName("linkTest");
+        linkTest.source(compile.getObjectFileDirectory().getAsFileTree().matching(new PatternSet().include("**/*.obj", "**/*.o").exclude("**/main.o")));
```

Comment:
If main.swift contains classes and functions that need to be unit tested, it should be included. What's the _actual_ problem here if we include main.swift? A linker error?